src/core/server/events/playerDeath.ts

```diff
@@ -3,10 +3,14 @@ import { playerFuncs } from '../extensions/Player';
 
 alt.on('playerDeath', handleDeath);
 
-function handleDeath(player: alt.Player, killer: alt.Player, weaponHash: any): void {
+function handleDeath(player: alt.Player, killer: alt.Entity, weaponHash: any): void {
     if (!player || !player.valid) {
         return;
     }
 
-    playerFuncs.set.dead(player, killer, weaponHash);
+    if (!killer || !(killer instanceof alt.Player) || !killer.valid) {
```

Comment:
This prevents respawning of a player if they're not killed by another player.

Reply:
I do this because killer is alt.Entity and just added additional check its alt.Player insance, but i see how u fixed in 2.0.1 much better than me.